### PR TITLE
Add and setup ESLint

### PR DESCRIPTION
**Pull request by matijs**
Add the Ultimaker ESLint configuration, set it up and add an empty `index.js` so there's something to be linted…

### CODE REVIEW

**Comment by ChrisTerBeke** [comment]
.gitlab-ci.yml is missing, so nothing is done in CI.

**Comment by ianpaschal** [comment]
I'm confused about the function of this repo. We don't use (and AFAIK don't need) stylelint...

**Comment by matijs** [comment]
@ianpaschal Not yet but we should. SCSS currently is all over the place.

**Comment by matijs** [comment]
@ChrisTerBeke soon…

**Comment by alcohol** [comment]
> We don't use (and AFAIK don't need) stylelint...

Can you elaborate on that? In my experience, there's a lot to be gained from having a coding/style guide and linting your projects accordingly. I have yet to hear good arguments for explicitly choosing to not do this.

**Comment by alcohol** [comment]
> set it up and add an empty `index.js`

Isn't `.eslintrc.js` a viable linting candidate for this already?

**Comment by ianpaschal** [comment]
> Can you elaborate on that? In my experience, there's a lot to be gained from having a coding/style guide and linting your projects accordingly. I have yet to hear good arguments for explicitly choosing to not do this.

> Not yet but we should. SCSS currently is all over the place.

My mistake. I was thinking of Prettier in my head. I'm against mixing linters per language. I've seen some really gross mashups of ESLint, TSLint, and Prettier all fighting with each other and spitting out some file rather than just setting up ESLint properly and auto fixing on save.

Indeed, a stylelint config for SCSS is fine, although I know very little about stylelint...

**Comment by ChrisTerBeke** [comment]
> @ChrisTerBeke soon…

Will that be done in this PR or another one?

**Comment by matijs** [comment]
@alcohol `.eslintrc.js` is ignored by default…

**Comment by matijs** [comment]
@ChrisTerBeke not a bad idea to add it here actually.

**Comment by Ghostkeeper** [comment]
The function of this repository is duplicated by the Ultimaker/Meta repository.

**Comment by ChrisTerBeke** [comment]
> The function of this repository is duplicated by the Ultimaker/Meta repository.

No, there will be an actual configuration in here that is published as an NPM package to include in all front-end projects.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+image: node:8
+
+stages:
+  - test
```
Comment (ChrisTerBeke) [inline .gitlab-ci.yml]: Should we also add a 'publish' stage here (like we do with eslint-config)?

Comment (ChrisTerBeke) [reply .gitlab-ci.yml]: 'deploy': https://github.com/Ultimaker/eslint-config/blob/master/.gitlab-ci.yml

**Comment by ianpaschal** [comment]
@Ghostkeeper This is similar to [@ultimaker/eslint-config](http://www.github.com/ultimaker/@ultimaker/eslint-config] in that it's a module that we can use so our linters actually make the style changes needed instead of them just purely living in documentation.

File: .gitlab-ci.yml (hunk 1 above)
Comment (matijs) [reply .gitlab-ci.yml]: Yes, but only once there is something to deploy :)

Comment (ChrisTerBeke) [reply .gitlab-ci.yml]: That's what the 'only' filter in the publish job is for. it only does the actual deploy when a tag is created.

Comment (matijs) [reply .gitlab-ci.yml]: Added, can I get a 👍 so this can be merged?

**Pull request merged**